### PR TITLE
UI Chart Scale Bar 2 (formerly #359)

### PR DESCRIPTION
**Pull request by alex40724**
This is the implementation of the original PR #359, which already got JF confirmation. I did not change anything related to the interface, so there is no need for an additional JF meeting at this stage.

Links to rules and description in factories:
https://github.com/ILIAS-eLearning/ILIAS/pull/572/files#diff-0c3b34e8901a0a9c29aec05f17067584
https://github.com/ILIAS-eLearning/ILIAS/pull/572/files#diff-353496e9f724608c4efd8dae098bafab

Related Feature is "Competence Reporting Panels":
https://www.ilias.de/docu/goto_docu_wiki_wpage_4314_1357.html

### CODE REVIEW

File: src/UI/Implementation/DefaultRenderer.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function getRendererFor($class) {
 	public function instantiateRendererFor($class) {
 		$renderer_class = $this->getRendererNameFor($class);
 		if (!class_exists($renderer_class)) {
-			throw new \LogicException("No rendered for '".$class."' found.");
+			throw new \LogicException("No renderer for '".$class."' found. (Renderer class $renderer_class)");
```
Comment (Amstutz) [inline src/UI/Implementation/DefaultRenderer.php:119]: +1

File: src/UI/templates/default/Chart/chart.less [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	min-width: 100px;
+	overflow: hidden;
+	margin: 0;
+	padding: 0;
+	list-style-type: none;
+}
+
+.il-chart-scale-bar > li {
+	float: left;
+	padding: 0 @il-chart-item-space;
```
Comment (Amstutz) [inline src/UI/templates/default/Chart/chart.less:11]: thx

File: templates/default/less/variables.less [hunk 3]
```diff
@@ -443,6 +444,17 @@
...
 
+//== Charts
+//
+//##
+
+@il-chart-item-active-bg: @brand-primary;
+@il-chart-item-active-color: white;
+@il-chart-item-inactive-bg: #e0e0e0;
+@il-chart-item-inactive-color: #909090;
+@il-chart-item-space: 2px;
```
Comment (Amstutz) [inline templates/default/less/variables.less:455]: Thx for this good selection of less variables. If the colors concept will be accepted, I will try to work it in here.

**Review by Amstutz — APPROVED** [review]
Hi Alex, those charts really look nice. I will try to work it the new color concept as soon as it gets accepted. I did not run the tests. If they pass, this can be merged.

**Comment by alex40724** [comment]
@Amstutz Thank you for taking time looking into this.

**Pull request merged**